common/src/chain/transaction/output/output_value.rs

```diff
@@ -16,35 +16,36 @@
 use serialization::{Decode, Encode};
 
 use crate::{
-    chain::tokens::{NftIssuance, TokenData, TokenIssuance, TokenTransfer},
+    chain::tokens::{NftIssuanceV0, TokenData, TokenId, TokenIssuanceV0, TokenTransfer},
     primitives::Amount,
 };
 
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize)]
 pub enum OutputValue {
     Coin(Amount),
-    Token(Box<TokenData>),
+    TokenV0(Box<TokenData>),
+    TokenV1(TokenId, Amount),
```

Comment:
Is this needed? Don't we have another output type now?

Reply:
yes that's important. It solves 2 problems
1. It makes the logic clean and explicit. Previously it was possible to create `Burn(TokenIssuance)` output which makes no sense and has to be handled.  `Transfer(TokenIssuance)`/`Transfer(TokenTransfer)` also looks weird. 
2. Now that issuing a token creates an account previous approach would break locked constrains, because output `LockThenTransfer(TokenIssuance)` locks utxo but not account.

With new OutputValue type transfers and burns are simple amount without any data that needs interpretation.

Reply:
I could potentially drop only TokenData::TokenIssuance but continue using TokenData::TokenTransfer, but I think that it's was a mistake in the first place to add such an enum, so I'd move away from it